### PR TITLE
Add prepare script to package.json

### PR DESCRIPTION
**Pull request by jake-low**
The `prepare` script is run automatically before `npm publish`, so having it run the build command might help to avoid issues like #61.

This only affects the workflow of package maintainers and not of end-users, so please feel free to reject this PR if you prefer a different method.

### CODE REVIEW

**Comment by Scrum** [comment]
@jake-low Hi, thanks so much for the input. Could you please figure out the reasons for the failure of tests ?

**Comment by jake-low** [comment]
I dug into this a bit.

This change has a side effect which is that the CI tests now try to build the package during their setup phase. Previously the CI system would only run the tests (using ts-node, I think).

It looks like `npm run build` fails under Node 10 (but succeeds under Node 12 and Node 14). The problem is that `tsup` uses worker threads which were introduced in Node 10.5 but were hidden behind the `--experimental-worker` flag until 11.7.

On my machine, using Node 10.14.0, I was able to fix this build error by running:

```
env NODE_OPTIONS=--experimental-worker npm run build
```

However, I immediately hit another error:

```
> rm -rf dist && tsup src/*.ts --minify

ReferenceError: TextDecoder is not defined
    at Object.<anonymous> (/Users/jake/Code/FOSS/posthtml/posthtml-parser/node_modules/tsup/dist/chunk.VLZO66V2.js:14172:19)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/Users/jake/Code/FOSS/posthtml/posthtml-parser/node_modules/tsup/dist/index.js:5:24)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
```

It appears that `tsup` uses the TextDecoder APIs that were added to the global namespace in Node 11 (they were available starting in Node 8 in the "util" module). Basically, I don't think `tsup` supports Node < 11.

Interestingly, I was also getting errors trying to run `npm run test` under Node 10 on my machine, even after reverting the change I proposed in this PR. I can't get AVA to run the tests; it fails with `TypeError: Cannot read property 'on' of null`. It appears [other people have encountered this too](https://github.com/avajs/ava/issues/2419) and updating from Node 10 to a later version fixed it for them. I'm not sure why this same workflow is working in the Github Actions CI system; it might have to do with differences between the exact Node version it's set up to use and the one I tested with.

In summary, this PR breaks the Node 10 tests because it's not possible to build the project currently under Node 10. It's possible that changing how the build works (perhaps removing `tsup`) could fix that. I'm not sure worrying about that is worthwhile, though: Node 10 is in maintenance mode and will reach end of life in two weeks (April 30), after which it won't receive any security updates.

**Comment by Scrum** [comment]
@jake-low Great investigation, let's drop support for node 10 and below. Can you make the necessary changes to this PR?

**Review by Scrum — APPROVED** [review]
